src/v2/BaseLoginRouter.js

```diff
@@ -29,12 +29,19 @@ import uiSchemaTransformer from './ion/uiSchemaTransformer';
 import i18nTransformer from './ion/i18nTransformer';
 import AppState from './models/AppState';
 import idx from 'idx';
+import { OKTA_STATE_TOKEN_KEY } from './view-builder/utils/Constants';
+
+const setStateHandleInStorage = (stateToken) => {
+  if(!sessionStorage.getItem(OKTA_STATE_TOKEN_KEY) && stateToken) {
+    sessionStorage.setItem(OKTA_STATE_TOKEN_KEY, stateToken);
+  }
+};
 
 const startLoginFlow = (settings) => {
+  let stateHandle = sessionStorage.getItem(OKTA_STATE_TOKEN_KEY);
```

Comment:
initialize() method stores the stateToken setStateHandleInStorage(this.settings.get('stateToken')) when widget first gets loaded. Now if stateToken remains valid then widget always gives precedence to existing token. Widget uses new token only in the cases where `/introspect` fails with existing token..

Now the issue arises for magic links, urls where enduser just paste the valid url with valid token on existing tab, in this case widget should be aware of using this url and new stateToken instead of existing one. How to do that ??